src/workerd/api/http.c++

```diff
@@ -1816,7 +1836,26 @@ jsg::Promise<jsg::Ref<Response>> fetchImplNoOutputLock(
   // If the jsRequest has a CacheMode, we need to handle that here.
   // Currently, the only cache mode we support is undefined, but we will soon support
   // no-cache and no-store. These additional modes will be hidden behind an autogate.
-  KJ_ASSERT(jsRequest->getCacheMode() == Request::CacheMode::NONE);
+  auto headerIds = ioContext.getHeaderIds();
+  const auto cacheMode = jsRequest->getCacheMode();
+  if(util::Autogate::isEnabled(util::AutogateKey::HTTP_REQUEST_CACHE)) {
+    switch(cacheMode) {
+      case Request::CacheMode::NOSTORE:
+      case Request::CacheMode::NOCACHE:
+      if(headers.get(headerIds.cacheControl) == kj::none) {
```

Comment:
Why would the `cacheControl`/`pragma` header already be set?

Reply:
iirc the user can set them.